packages/core/src/rest/Youtube.ts

```diff
@@ -115,19 +116,28 @@ export async function trackSearchByString(query: string, omitStreamId?: string,
     results.items as ytsr.Video[],
     item => (!omitStreamId || item.id !== omitStreamId)
   ) as ytsr.Video;
-  const topTrackInfo = await ytdl.getInfo(topTrack.url);
-  const formatInfo = ytdl.chooseFormat(topTrackInfo.formats, { quality: 'highestaudio' });
-  const segments = await SponsorBlock.getSegments(topTrack.id);
-  return {
-    source: sourceName,
-    id: topTrack.id,
-    stream: formatInfo.url,
-    duration: parseInt(topTrackInfo.videoDetails.lengthSeconds),
-    title: topTrackInfo.videoDetails.title,
-    thumbnail: topTrack.bestThumbnail.url,
-    format: formatInfo.container,
-    skipSegments: segments
-  };
+
+  try{
+    const topTrackInfo = await ytdl.getInfo(topTrack.url);
+    const formatInfo = ytdl.chooseFormat(topTrackInfo.formats, { quality: 'highestaudio' });
+    const segments = await SponsorBlock.getSegments(topTrack.id);
+  
+    return {
+      source: sourceName,
+      id: topTrack.id,
+      stream: formatInfo.url,
+      duration: parseInt(topTrackInfo.videoDetails.lengthSeconds),
+      title: topTrackInfo.videoDetails.title,
+      thumbnail: topTrack.bestThumbnail.url,
+      format: formatInfo.container,
+      skipSegments: segments
+    };
+  }
+    catch (e){
+      logger.error(`Warning: topTrack.url is undefined, removing song`);
+      return undefined;
```

Comment:
I think I will better to throw an error instead of `return undefined`

Reply:
Yes, you should throw an error here, and then we can catch it in the thunk and dispatch an error action.